### PR TITLE
Pep8 pylint switch

### PR DESCRIPTION
**Pull request by marcospri**
Replace pep8 by pylint

- Changes in make and tox files
- Removed `#noqa` annotations for pep8
- Pending ignored pylint rules tracked in https://github.com/hypothesis/h/issues/6772

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -65,7 +64,7 @@ sql: python
 	@tox -qe dockercompose -- exec postgres psql --pset expanded=auto -U postgres
 
 .PHONY: lint
-lint: backend-lint frontend-lint
+lint: frontend-lint backend-lint
```
Comment (marcospri) [inline Makefile:67]: Swapped to run the fastest first

**Review by jon-betts — COMMENTED** [review]
This works for me, but is unfortunately 1m20s vs 20s, but I guess that's the price we pay for the thoroughness.

I was looking through the Jenkins file and I noticed the "backend" tasks are running the combined versions of `tests` and `lint` where they should probably be doing `backend-tests` etc. With the `frontend-tests` moved to the frontend step.

There's also some weirdness that the functests run the linting? This doesn't make much sense to me, but I'll bet there's a reason for it.

**Comment by marcospri** [comment]
> There are still a number of plain `noqa` marks left in the code.

Removed those, except a few from pycodestyle that are left in the migrations. Added the exact rule they are disabling to those.

**Comment by marcospri** [comment]
> This works for me, but is unfortunately 1m20s vs 20s, but I guess that's the price we pay for the thoroughness.
> 
> I was looking through the Jenkins file and I noticed the "backend" tasks are running the combined versions of `tests` and `lint` where they should probably be doing `backend-tests` etc. With the `frontend-tests` moved to the frontend step.

I can see this on Jenkins (but maybe there's some import mechanism that I'm missing?) On Jenkins I only see `sh 'cd /var/lib/hypothesis && tox -e functests'` whcih should be ok.

On the github actions the backend uses tox (backend only)  and the frontend uses make with the `frontend-` targets

> There's also some weirdness that the functests run the linting? This doesn't make much sense to me, but I'll bet there's a reason for it.

Moved the linting close to checkformatting and left the functests on their own.

**Review by jon-betts — APPROVED** [review]
I couldn't approve this before because Github had gone nuts, but this looks good to me!

**Pull request merged**